cloudferrylib/os/actions/transport_instance.py

```diff
@@ -57,23 +58,30 @@ def run(self, info=None, **kwargs):
             utl.INSTANCES_TYPE: {
             }
         }
-
-        # Get next one instance
-        for instance_id, instance in info[utl.INSTANCES_TYPE].iteritems():
-            instance = self._replace_user_ids(instance)
-
-            one_instance = {
-                utl.INSTANCES_TYPE: {
-                    instance_id: instance
+        processing_vms = kwargs.get('processing_vms', [])
+        try:
+            # Get next one instance
+            for instance_id, instance in info[utl.INSTANCES_TYPE].iteritems():
+                instance = self._replace_user_ids(instance)
+
+                one_instance = {
+                    utl.INSTANCES_TYPE: {
+                        instance_id: instance
+                    }
                 }
-            }
-            one_instance = self.deploy_instance(self.dst_cloud, one_instance)
-
-            new_info[utl.INSTANCES_TYPE].update(
-                one_instance[utl.INSTANCES_TYPE])
+                one_instance = self.deploy_instance(self.dst_cloud,
+                                                    one_instance)
+                processing_vms.append(
+                    one_instance[utl.INSTANCES_TYPE].keys()[0])
```

Comment:
It is not so clear what `[utl.INSTANCES_TYPE].keys()[0]` returns. Clarify please.